Framework/Geometry/src/Rendering/GeometryTriangulator.cpp

```diff
@@ -58,23 +59,37 @@ Kernel::Logger g_log("GeometryTriangulator");
 } // namespace
 
 GeometryTriangulator::GeometryTriangulator(const CSGObject *obj)
-    : m_isTriangulated(false), m_nFaces(0), m_nPoints(0) {
-  m_obj = obj;
+    : m_isTriangulated(false), m_nFaces(0), m_nPoints(0), m_csgObj(obj) {
 #ifdef ENABLE_OPENCASCADE
   m_objSurface = nullptr;
 #endif
 }
 
+GeometryTriangulator::GeometryTriangulator(const MeshObject *obj)
+    : m_isTriangulated(false), m_meshObj(obj) {
+}
+
 GeometryTriangulator::~GeometryTriangulator() {}
 
 void GeometryTriangulator::triangulate() {
 #ifdef ENABLE_OPENCASCADE
   if (m_objSurface == nullptr)
     OCAnalyzeObject();
 #endif
+  if (m_meshObj) {
+    generateMesh();
+    m_nPoints = m_meshObj->numberOfVertices();
+    m_nFaces = m_meshObj->numberOfTriangles();
+    m_points = m_meshObj->getVertices();
+    m_faces = m_meshObj->getTriangles();
+  }
   m_isTriangulated = true;
 }
 
+void GeometryTriangulator::generateMesh() {
```

Comment:
@KarlPalmen The changes in this PR consolidate the GeometryHandlers/Generators. The `GeometryTriangulator` is now responsible for generating the faces and vertices for triangulated surfaces. I have added this method to replace `MeshGeometryGenerator::Generate` which was previously unimplemented. We can discuss a strategy for added the mesh generation in if you want but I think it is straightforward and easy to see now that the code for all of this has been significantly compressed.